aiida/orm/nodes/data/code/remote.py

```diff
@@ -0,0 +1,168 @@
+        Checks whether the executable exists on the remote computer if a transport can be opened to it. This method
+        is intentionally not called in ``_validate`` as to allow the creation of ``Code`` instances whose computers can
+        not yet be connected to and as to not require the overhead of opening transports in storing a new code.
+
+        :raises `~aiida.common.exceptions.ValidationError`: if no transport could be opened or if the defined executable
+            does not exist on the remote computer.
+        """
+        try:
+            with override_log_level():  # Temporarily suppress noisy logging
+                with self.computer.get_transport() as transport:
+                    file_exists = transport.isfile(self.filepath_executable)
+        except Exception:  # pylint: disable=broad-except
+            raise exceptions.ValidationError(
+                'Could not connect to the configured computer to determine whether the specified executable exists.'
+            )
+
+        if not file_exists:
+            raise exceptions.ValidationError(
+                f'The provided remote absolute path `{self.filepath_executable}` does not exist on the computer.'
+            )
+
+    def can_run_on_computer(self, computer: Computer) -> bool:
+        """Return whether the code can run on a given computer.
+
+        :param computer: The computer.
+        :return: ``True`` if the provided computer is the same as the one configured for this code.
+        """
+        type_check(computer, Computer)
+        return computer.pk == self.computer.pk
+
+    def get_executable_run_line(self) -> str:
+        """Return the line for the submission script that should be executed to run the code.
+
+        :return: The run line to be executed in the submission script.
+        """
+        return self.filepath_executable
+
+    @property
+    def full_label(self) -> str:
+        """Return the full label of this code.
+
+        The full label can be just the label itself but it can be something else. However, it at the very least has to
+        include the label of the code.
+
+        :return: The full label of the code.
+        """
+        return f'{self.label}@{self.computer.label}'
+
+    @property
+    def filepath_executable(self) -> pathlib.PurePath:
+        """Return the absolute filepath of the executable that this code represents.
+
+        :return: The absolute filepath of the executable.
+        """
+        return pathlib.PurePath(self.base.attributes.get(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE))
+
+    @filepath_executable.setter
+    def filepath_executable(self, value: str) -> None:
+        """Set the absolute filepath of the executable that this code represents.
+
+        :param value: The absolute filepath of the executable.
+        """
+        type_check(value, str)
+
+        if not pathlib.PurePath(value).is_absolute():
+            raise ValueError('the `filepath_executable` should be absolute.')
```

Comment:
Good point. I now vaguely remember an issue being opened at some point to support this, but I cannot find it. Not sure what happened to it. Not sure how we could easily validate this though. If we allow this, it could be that the executable resolving relies on bash code that is part in the `Computer` prepend/append text.

Reply:
> If we allow this, it could be that the executable resolving relies on bash code that is part in the Computer prepend/append text.

Yes, I think prepend text in `Computer` is not a problem, it can be read here for RemoteCode just as the prepend text of `Code`. But it really not easy to separate the `module load` part or maybe a manually re-assigned `PATH` environment variable. Probably, the only way is to separate the 'path modified' commands from prepend text, but it increases the burden when setting computer and code.